xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/XWikiIcon.vue

```diff
@@ -39,27 +44,76 @@
 
 
 <script>
+const iconCache = {};
+
 export default {
 
   name: "XWikiIcon",
 
+  props: {
+    iconDescriptor: Object,
+  },
+
+  data () {
+    return {
+      remoteIconDescriptor: undefined,
+    };
+  },
+
   computed: {
     iconSetType () {
-      if (this.iconDescriptor.iconSetType) { return this.iconDescriptor.iconSetType; }
-      else if (this.iconDescriptor.cssClass) { return "FONT"; }
-      else if (this.iconDescriptor.url) { return "IMAGE"; }
+      if (this.iconDescriptor?.iconSetType) { return this.iconDescriptor?.iconSetType; }
+      else if (this.iconDescriptor?.cssClass) { return "FONT"; }
+      else if (this.iconDescriptor?.url) { return "IMAGE"; }
+      else if (this.iconDescriptor?.name) { return "REMOTE"; }
       return "";
     },
   },
 
-  props: {
-    iconDescriptor: Object,
+  methods: {
+    async fetchRemoteIconDescriptor (iconName) {
+      try {
+        const iconURLQuery = `?media=json&name=${iconName}`;
+        const iconURL = `/xwiki/rest/wikis/${XWiki.currentWiki}/iconThemes/icons${iconURLQuery}`;
+        const response = await fetch(iconURL);
```

Comment:
Sadly, fetch is not supported by IE11 (see https://caniuse.com/?search=fetch) but we are still supporting IE11 as per our browser support strategy (see https://dev.xwiki.org/xwiki/bin/view/Community/SupportStrategy/BrowserSupportStrategy#HSupportedBrowsers)